libs/server/Resp/RespServerSession.cs

```diff
@@ -292,64 +294,52 @@ private void ProcessMessages()
         {
             // #if DEBUG
             // logger?.LogTrace("RECV: [{recv}]", Encoding.UTF8.GetString(new Span<byte>(recvBufferPtr, bytesRead)).Replace("\n", "|").Replace("\r", ""));
+            // Debug.WriteLine($"RECV: [{Encoding.UTF8.GetString(new Span<byte>(recvBufferPtr, bytesRead)).Replace("\n", "|").Replace("\r", "")}]");
             // #endif
 
             var _origReadHead = readHead;
 
             while (bytesRead - readHead >= 4)
             {
-                // NOTE: Possible optimization: Don't parse if only parsing AUTH and not authenticated.
-                ReadOnlySpan<byte> specificErrorMessage = default;
-                RespCommand cmd = ParseCommand(out int count, recvBufferPtr + readHead, out bool success, ref specificErrorMessage);
+                // First, parse the command, making sure we have the entire command available
+                // We use endReadHead to track the end of the current command
+                // On success, readHead is left at the start of the command payload for legacy operators
+                var cmd = ParseCommand(out int count, out bool commandReceived);
 
```

Comment:
OK here are the (end)readHead comments I wanted :) Nice to have something at the point of declaration though

Reply:
added